downstairs/src/region.rs

```diff
@@ -905,6 +908,43 @@ impl Region {
             result??;
         }
 
+        if cfg!(feature = "omicron-build") {
```

Comment:
Using  `if cfg!(..)` weirds me out because the code is still compiled even if the feature is disabled; I guess the `ioctl` code builds just fine on MacOS, but I'd rather not!

What about something like this?

```rust
#[cfg(feature = "omicron-build")]
{
    #[cfg(not(target_os = "illumos"))]
    compile_error!("cannot use FIOFFS on non-illumos systems");
    
    // .. normal code continues below
```